planar/planar_test.go

```diff
@@ -0,0 +1,46 @@
+package planar
+
+import (
+	"strconv"
+	"testing"
+)
+
+func TestSlope(t *testing.T) {
+	type tcase struct {
+		line    [2][2]float64
+		m, b    float64
+		defined bool
+	}
+
+	fn := func(t *testing.T, tc tcase) {
+		t.Parallel()
+		gm, gb, gd := Slope(tc.line)
+		if tc.defined != gd {
+			t.Errorf("sloped defined, expected %v got %v", tc.defined, gd)
+			return
+		}
+		// if the slope is not defined, line is verticle and m,b don't have good values.
+		if !tc.defined {
+			return
+		}
+		if tc.m != gm {
+			t.Errorf("sloped, expected %v got %v", tc.m, gm)
+
+		}
+		if tc.b != gb {
+			t.Errorf("sloped intercept, expected %v got %v", tc.b, gb)
+		}
+	}
+	tests := []tcase{
+		{
+			line:    [2][2]float64{{0, 0}, {10, 10}},
+			m:       1,
+			b:       0,
+			defined: true,
```

Comment:
With all the test framework in place, you'll want to add some more tests. For instance, this won't catch a case where x/y get swapped.

Reply:
Yeah, I don't have very good tests in there right now.  I was still in the process of converting over the code, and did not have time to add all the test cases. This was just to get the framework for the test in place, to make it easy to add more.